src/django_components/apps.py

```diff
@@ -5,4 +5,9 @@ class ComponentsConfig(AppConfig):
     name = "django_components"
 
     def ready(self) -> None:
+        # This is the code that gets run when user adds django_components to Django's INSTALLED_APPS
+        #
+        # TODO - Allow only if autodiscovery is true?
+        #        Because right now it's kinda weird that we import the COMPONENTS.libraries
+        #        Even if autodiscovery is off.
         self.module.autodiscover()
```

Comment:
I think it makes sense to make the `autodiscover` function as part of the public API, so that people can possibly run it when they need to. However, that invites a discussion on how `autodiscover` should behave:

So autodiscover currently actually imports from 3 places:
1. Component modules - `[app_name]/components.py` for each app in the project
2. Component dirs - As defined by `Loader.get_dirs`, which takes the `STATICFILES_DIRS`, and for each one tries to look in a couple of places
3. Libraries - Modules listed in `COMPONENTS.libraries`.

How things worked up to now is that:
- 1. and 2., so `[app_name]/components.py` and `STATICFILES_DIRS`, were loaded only if `AUTODISCOVER` setting was ON.
- 3., the `COMPONENTS.libraries`, were ALWAYS imported.

This feels a bit random, and raises a few questions:

1. Should we split the logic or rename the function? The "auto*discovery*" is really only the part where we search `STATICFILES_DIRS` for python files. The importing of `COMPONENTS.libraries` and `[app_name]/components.py` is more like "import_libraries" or "import_modules"

2. Should we drop support for automatically searching for `[app_name]/components.py`, and let users set that manually via `COMPONENTS.libraries`? Or we can externalize it as a helper function that generates a list of paths like `[app_name]/components.py`, so users can then pass the results to `COMPONENTS.libraries`.

	Note that the `[app_name]/components.py` feature is NOT documented currently.

	```py
	from django_components.utils import gen_app_paths
	COMPONENTS = {
		"libraries": [
			# Generates `[app_name]/components.py` for each app
			*gen_app_paths("components"),

			# Other 
			"my_app.some.path",
			...
		],
	}
	```

5. How should we handle when `AUTODISCOVER` is ON / OFF?
	First of all, I think we should remove `[app_name]/components.py` (1. from the beginning of this comment) and let users set the same thing via `COMPONENTS.libraries`
	
	Then, there would be only imports via `COMPONENTS.libraries` (3.) and `STATICFILES_DIRS` (2.).

	- If user wanted to turn off imports via `COMPONENTS.libraries`, they could just set it to an empty list
	- And if user wanted to turn off imports from `STATICFILES_DIRS`, they would set `AUTODISCOVER` to `False`.

	So that means that `AUTODISCOVER` would only decide whether to search for python files in `STATICFILES_DIRS`.

---

So all-in-all, my approach would be:
- The `autodiscover.py` file would export two functions: `autodiscover` and `import_libraries`

-  `autodiscover` function would have signature
	```py
	def autodiscover(
	    map_module_names: Optional[Callable[[str], str]] = None,
	) -> List[str]:
	```
	Where:
	- `map_module_names` is renamed from `map_components_modules`, and it is kept as we use that in some of the tests.
	- `autodiscover` function returns a list of module names that were imported
	- `autodiscover` function only searches `STATICFILES_DIRS` for python files, and does NOT import from `COMPONENTS.libraries`, nor from `[app_name]/components.py`.
	- `autodiscover` function would run inside the [`ready` hook](https://github.com/EmilStenstrom/django-components/pull/533/files#diff-2e6e554ee041581d50482b56d52761e5d70e4d1c58dbe75352ac2e72d49b3151R13) only if `AUTODISCOVER` setting is ON. However, user could import the function and run it themselves whenever.

-  `import_libraries` function would have signature
	```py
	def import_libraries(
	    map_module_names: Optional[Callable[[str], str]] = None,
	) -> List[str]:
	```
	Where:
	- `map_module_names` is added to have the same functionality as `autodiscover`
	- `import_libraries` function returns a list of module names that were imported
	- `import_libraries` only imports from `COMPONENTS.libraries`, nothing else.
	- `import_libraries` would run inside the [`ready` hook](https://github.com/EmilStenstrom/django-components/pull/533/files#diff-2e6e554ee041581d50482b56d52761e5d70e4d1c58dbe75352ac2e72d49b3151R13) ALWAYS. However, user could import the function and run it themselves whenever.

So [src/django_components/apps.py](https://github.com/EmilStenstrom/django-components/pull/533/files#diff-2e6e554ee041581d50482b56d52761e5d70e4d1c58dbe75352ac2e72d49b3151) would look like this:
```py
from django.apps import AppConfig

class ComponentsConfig(AppConfig):
    name = "django_components"

    def ready(self) -> None:
        # This is the code that gets run when user adds django_components to Django's INSTALLED_APPS
	from django_components.app_settings import app_settings
	from django_components.autodiscover import autodiscover, import_libraries	

	import_libraries()

	if app_settings.AUTODISCOVER:
	        autodiscover()
```

Reply:
UPDATE: Updated as described above